test/algorithm_test_cases.h

```diff
@@ -0,0 +1,181 @@
+/**
+ * @author      Yanqing Wu
+ * @email       meet.yanqing.wu@gmail.com
+ * @create date 2023-04-02 22:32:23
+ * @modify date 2023-04-03 01:02:09
+ * @desc Sharing Test
+ * 
+ * https://github.com/google/googletest/blob/main/docs/advanced.md#sharing-resources-between-tests-in-the-same-test-suite
+ */
+
+#ifndef FG_TEST_ALGORITHM_TEST_CASES_H_
+#define FG_TEST_ALGORITHM_TEST_CASES_H_
+
+// gtest
+#include <gtest/gtest.h>
+// std
+#include <sstream>
+#include <vector>
+// local
+#include "game.h"
+
+/**
+ * @brief We are testing against the WIN/LOSS/UNKNOWN result.
+ * With result=WIN, we are also testing against the winning move (only for
+ * certain cases).
+ *
+ * Simple test: board less than and equal to 3x3,
+ * Midium test: board size between 4x4 and 7x7,
+ * Large  test: board size between 8x8 and 10x10; or too many empty cells
+ */
+
+using TestContainer = std::vector<std::pair<std::string, short>>;
+
+unsigned short getIndexFromName(const char* x) {
+  std::stringstream s(x);
+  char c;
+  while (c != '_') {
+    s >> c;
+  }
+  unsigned short y;
+  s >> y;
+  // std::cout << x << std::endl;   // for the very weird bug in minimax-ab-tt and negamax-ab-tt medium.1 and medium.3
```

Comment:
@Samarium150 I've encountered the weirdest bug so far.
I parse the test name to index, e.g. `SIMPLE_16` becomes an integer 16. This works well for 200+ test cases, except for `MEDIUM_1` and `MEDIUM_3` in `MinimaxAlphaBetaTTTest` and `NegamaxAlphaBetaTTTest`.
This is weird because `MEDIUM_1` and `MEDIUM_3` parsing works for other test classes, but it just won't work for the two I mentioned.

Reply:
I now hardcode for the exceptions if you look at the code. I just don't understand the bug.

Reply:
look into it after MCTS